source/types/trinary.cpp

```diff
@@ -84,9 +84,81 @@ tritsToBytes(const Trits& trits) {
 
 Trits
 bytesToTrits(const std::vector<int8_t>& bytes) {
-  Types::BigInt decimal;
-  decimal.fromBytes(bytes);
-  return decimal.toTrits();
+  if (!bytes.size()) {
+    return std::vector<int8_t>(TritHashLength, 0);
+  }
+  int8_t sign = bytes[0] >= 0 ? 1 : -1;
+  size_t nr_bytes = bytes.size();
+  size_t size = (nr_bytes + 3) >> 2;
+  std::vector<uint32_t> div(size, 0);
+  uint32_t dw = sign == 1 ? 0 : (0xffffffff << 8);
+  for (size_t i = 0; i < nr_bytes; ++i) {
+    dw |= static_cast<uint8_t>(bytes[i]);
+    if (!((nr_bytes - i - 1) & 0x3)) {
+      div[(nr_bytes - i - 1) >> 2] = dw;
+      dw = 0;
+    }
+    dw <<= 8;
+  }
+
+  // negate two's complement if sign is negative
+  if (sign == -1) {
+    size_t i = 0;
+    for (; i < div.size(); ++i) {
+      div[i] = ~div[i] + 1;
+      if (div[i]) {
+        ++i;
+        break;
+      }
+    }
+    for (; i < div.size(); ++i) {
+      div[i] = ~div[i];
+    }
+  }
+
+  std::vector<int8_t> trits;
+
+  // strip leading zeroes
+  while (!div.back()) {
+    div.pop_back();
```

Comment:
@mistersandman Maybe I'm wrong but calling pop_back() on an empty vector is undefined behavior and this can happen if bytesToTrits is given only zero bytes. This could lead to memory corruption. Would you care to check ? Thanks in advance

Reply:
In abc6c4956eda75c2772f31b7affaf850fcc71a22, I added some trits and bytes conversion tests. Line 143 of trinary_test.cpp is causing segmentation fault.

Reply:
You are perfectly right! Already `div.back()` exposes undefined behavior in the case of all bytes being zero, thanks for catching this!

PR #224 fixes the issue, by never actually changing the size of `div`, but only working with indexes into `div`.